tests/performance_tests/log_reader.py

```diff
@@ -70,3 +78,51 @@ def scrapeLog(total, path):
                         total.updateTotal(int(value[2]), 0, 0, 0, 0, int(v2Logging[0]))
                 else:
                     print("Error: Unknown log format")
+
+def findPrunedRangeOfInterest(data: chainData, numAddlBlocksToDrop=0):
```

Comment:
This method can be eliminated in favor of constructing the list to iterate in place in the calling function in one line:
```
filtered = filter(lambda block: block.transactions != 0, data.blockLog)
```
and then iterating it as:
```
for first, second in zip(*[iter(filtered[numAddlBlocksToDrop:-numAddlBlocksToDrop])]*2):
    consecutiveBlockTps.append(first.transactions + second.transactions)
```
which has the benefit of not double-counting transactions as the current code does.

The `zip(*[iter(sequenceToIterate)]*n)` idiom to step through an iterable in groups of `n` items is documented in the Python documentation for the `zip()` function, so it's fairly common to see in code which doesn't use `more_itertools`.  The additional wrinkle here is just slice notation to chop off additional blocks to drop at the beginning and end of the list.

Generally speaking, you don't usually need to use container indexes in Python code.